eventmesh-retry/eventmesh-retry-pulsar/src/main/java/org/apache/eventmesh/retry/pulsar/PulsarRetryStrategyImpl.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.eventmesh.retry.pulsar;
+
+import org.apache.eventmesh.api.SendCallback;
+import org.apache.eventmesh.api.SendResult;
+import org.apache.eventmesh.api.exception.OnExceptionContext;
+import org.apache.eventmesh.api.producer.Producer;
+import org.apache.eventmesh.common.protocol.http.common.ProtocolKey;
+import org.apache.eventmesh.common.protocol.http.common.ProtocolKey.ClientInstanceKey;
+import org.apache.eventmesh.retry.api.conf.RetryConfiguration;
+import org.apache.eventmesh.retry.api.strategy.RetryStrategy;
+
+import org.apache.pulsar.client.util.RetryMessageUtil;
+
+import java.util.Objects;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.core.builder.CloudEventBuilder;
+
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class PulsarRetryStrategyImpl implements RetryStrategy {
+    @Override
+    public void retry(final RetryConfiguration configuration) {
+        sendMessageBack(configuration);
+    }
+    
+    @SneakyThrows
+    private void sendMessageBack(final RetryConfiguration configuration) {
+        CloudEvent event = configuration.getEvent();
+        String topic = configuration.getTopic();
+        String consumerGroupName = configuration.getConsumerGroupName();
+        String retryTopicName = consumerGroupName + RetryMessageUtil.RETRY_GROUP_TOPIC_SUFFIX;
```

Comment:
I am not familiar with Pulsar, but I have two questions about your code:
1 Pulsar's message retry feature is turned off by default and needs to be turned on manually, how did you turn it on?
2 I don't understand how you realize that Pulsar automatically retries messages in the "xxx-RETRY" queue. Could you explain?

Reply:
Hi @pandaapo the retry feature is not turned on. For turning it on can the **_retryAsynchronously(Supplier<CompletableFuture<T>> supplier, Backoff backoff, ScheduledExecutorService scheduledExecutorService, CompletableFuture<T> callback)_** method of **_RetryUtil_** class of **_pulsar-client-2.11.1.jar_** library be considered as it provides with the method **_scheduledExecutorService.execute()_** to schedule and execute a retry. Can you guide?
![image](https://github.com/apache/eventmesh/assets/96522794/aba7d7a6-2541-44c4-b2bc-1b8ba103ad81)

Reply:
@HarshSawarkar 
RocketMQ can automatically retry messages in the "% RETRY%..."queue, so `RocketMQRetryStrategyImpl` utilizes RocketMQ's retry queue feature and sends failed messages to a queue with the prefix "% RETRY%".
When it comes to Pulsar, we also need Pulsar to automatically retry failed messages. Currently your code does not implement this feature.

Reply:
Could you provide some implementation ideas on how to implement the feature for automatically retrying failed messages of Pulsar?

Reply:
@HarshSawarkar I wish I could give more useful help, but as I said earlier, I'm not familiar with Pulsar.

Reply:
> you'll need to resend the messages through EventMesh. You can achieve this using an in-memory queue and a timer.

This suggestion is incorrect.  Because the task of this PR is to implement retry in Pulsar without relying on EventMesh's memory. Otherwise, the PR would be redundant since EventMesh retries within the memory by default.